core/src/main/java/org/apache/iceberg/BaseMetastoreCatalog.java

```diff
@@ -301,4 +305,15 @@ protected static String fullTableName(String catalogName, TableIdentifier identi
 
     return sb.toString();
   }
+
+  private synchronized MetricsReporter metricsReporter() {
```

Comment:
nit: I don't think this needs to be synchronized

Reply:
ok